### PR TITLE
[master] Print asset support added

### PR DESCRIPTION
**Pull request by bstrx**
***UPD: tests added!***
To implement trimming of poster 19x27 we need to send a position of barcode to Profit. With guys from profit we decided to do it more abstract so i introduced PrintAsset for InkRouter sdk.
For specific product side we can specify several Print Assets.
 
Each Print Asset has:
 1. type - for now it will be only BARCODE, but we can introduce Image type or Text type later (but it will require some Value field)
 2. positionX - the x coordinate in inches
 3. positionY - the y coordinate in inches
 4. width - width in inches of the asset block where profit will place barcode
 5. height - in inches
 6. rotation - f.e. -90 - angle for asset to make it vertical, horizontal etc.

the xml will look something like this (search for '<print_assets>'):
```xml
<?xml version="1.0" encoding="UTF-8"?>
<order_info>
    <header_info>
        <from_domain>overnightprints.com</from_domain>
        <from_identity>overnightprints.com</from_identity>
    </header_info>
    <po_info>
        <agent_id>LisaL</agent_id>
        <currency>USD</currency>
    </po_info>
    <order>
        <print_customer_invoice>24018307</print_customer_invoice>
        <ts_created>2015-11-12T01:28:30-08:00</ts_created>
        <delivery_date>2015-11-20T12:00:00-08:00</delivery_date>
        <priority>0</priority>
        <shipping_fee>22.53</shipping_fee>
        <product_discounts>0</product_discounts>
        <shipping_discounts>0</shipping_discounts>
        <vendor_id>0</vendor_id>
        <contact>
            <name>test test</name>
            <phone>9999999999</phone>
            <email>no@email.com</email>
        </contact>
        <requester>
            <name>Jaisor Prints</name>
            <contract>STANDARD</contract>
            <pay_term>FREE</pay_term>
        </requester>
        <ship_address>
            <attention>test test</attention>
            <street_address>tes</street_address>
            <city>test</city>
            <state>KY</state>
            <zip>41001</zip>
            <country>US</country>
        </ship_address>
        <order_items>
            <order_item>
                <print_group_id>pg56445a2dc6674</print_group_id>
                <product_type>4.25x5.5 postcards</product_type>
                <paper_type>14PT</paper_type>
                <quantity>1000</quantity>
                <region_size>US</region_size>
                <cost>159.95</cost>
                <attributes>
                    <scalar_boolean_attribute>
                        <type>ROUNDING</type>
                        <value>true</value>
                    </scalar_boolean_attribute>
                    <scalar_boolean_attribute>
                        <type>LABELING</type>
                        <value>false</value>
                    </scalar_boolean_attribute>
                </attributes>
                <sides>
                    <side>
                        <page_number>0</page_number>
                        <file_url>http://onp.dev/tmp/m_opstore3/optmp9630//fhi_v2_5644599e35df17.05751544.jpg.tif</file_url>
                        <file_hash>b3f2b994d6ba760f76ddfa907a9f2c8e</file_hash>
                        <coating>FULL</coating>
                        <orientation>Landscape</orientation>
                        <print_assets>
                            <print_asset>
                                <position_x>18.51</position_x>
                                <position_y>8</position_y>
                                <width>6.25</width>
                                <height>1.5</height>
                                <type>BARCODE</type>
                                <rotation>-90</rotation>
                            </print_asset>
                        </print_assets>
                    </side>
                    <side>
                        <page_number>1</page_number>
                        <file_url>http://onp.dev/tmp/m_opstore3/optmp9630//bhi_v2_5644599ee43520.07710513.jpg.tif</file_url>
                        <file_hash>b011765e36596157730292556e34bf67</file_hash>
                        <coating>SPOT</coating>
                        <orientation>Landscape</orientation>
                        <spot_uv_file_url>http://onp.dev/tmp/m_opstore3/optmp9630//bhi_v2_5644599ee43520.07710513.jpg.tif.uv</spot_uv_file_url>
                        <spot_uv_file_hash>a519d6e57a8a031a988abbe989632e70</spot_uv_file_hash>
                    </side>
                </sides>
            </order_item>
        </order_items>
    </order>
    <print_customer_id>overnightprints.com</print_customer_id>
</order_info>

```

### CODE REVIEW

**Comment by bstrx** [comment]
@richardfullmer,
@maria-p 
review please

File: src/InkRouter/Models/PrintAsset.php [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+        if (isset($this->positionY)) {
+            $writer->writeElement('position_y', $this->positionY);
+        }
+
+        if (isset($this->width)) {
+            $writer->writeElement('width', $this->width);
+        }
+
+        if (isset($this->height)) {
+            $writer->writeElement('height', $this->height);
```
Comment (richardfullmer) [inline src/InkRouter/Models/PrintAsset.php]: Are these all pixels as the unit?

File: src/InkRouter/Models/PrintAsset.php [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+        $writer->openMemory();
+
+        $writer->startElement('print_asset');
+
+        if (isset($this->positionX)) {
+            $writer->writeElement('position_x', $this->positionX);
+        }
+
+        if (isset($this->positionY)) {
+            $writer->writeElement('position_y', $this->positionY);
```
Comment (richardfullmer) [inline src/InkRouter/Models/PrintAsset.php]: What's the origin point of your x/y coordinate system?  Upper left?  Lower left?  We should probably define that in the XML so there's no ambiguity.

Comment (bstrx) [inline src/InkRouter/Models/PrintAsset.php]: i use upper-left coordinates, but i'm not sure what is the best way to specify it (i'll look for standard or something)

File: src/InkRouter/Models/PrintAsset.php (hunk 1 above)
Comment (bstrx) [inline src/InkRouter/Models/PrintAsset.php]: i think we better use inches everywhere. I'll specify that in comments

File: src/InkRouter/Models/PrintAsset.php (hunk 2 above)
Comment (bstrx) [inline src/InkRouter/Models/PrintAsset.php]: also i talked with Profit and it seems that they won't be able to implement support of different origin points in time (if it is expected to be done along with introducing new property)

**Comment by bstrx** [comment]
@richardfullmer i've update the unit tests

**Pull request merged**